### DISCUSSION TITLE
Issues / Contribution

### DISCUSSION THREAD

**Discussion by xeniode** [Mobile-Artificial-Intelligence/maid#123] [General]
Hey there,

I was going to make my own mobile app for ollama with the same features. I decide to see if any exist, and alas I found this. I would like to help. I saw at the bottom of the README there were a few call outs for help, but they do not have much details. And I do not nee anything in issues. Let me know what would be needed help on and I will jump in. 

Thanks! and awesome job.

**Comment by danemadsen** [comment]
Thanks, Any help is appreciated. The main thing that i need someone else to do is get it working on apple platforms as i don't have one of their systems. As for anything else, I'm not really sure. I'm happy to take suggestions on how to improve the app.

I don't really have any solid plans for where I'm going with the app I'm really just programming it for fun.

**Reply by HenryDev** [reply]
I came across this project after reading this [blog post on AI chat apps](https://mazzzystar.github.io/2023/11/16/ai-girlfriend-product/). I don't know how much fun you get from your day job, but I think this app has the earning potential to replace it. Some ideas off the top of my head:
1. out of the box functionality for hosted APIs like chatgpt, so users can easily compare custom vs standard models
2. support for managed model offered by cloud platforms, this can turn the app into a centralized hub 
3. auth'd sessions / sign in with x to share sessions across platforms, and enable copy paste, upload download, share with contacts, etc.
4. group chats? the possibilities are endless

**Reply by danemadsen** [reply]
Thank you that's high praise. 

Integrating other API's is straight forward enough I can get that done pretty easily. As for managing models on a cloud platform that's not really in the scope of maids purpose, anyone that wants to host a model like you describe can already host an Ollama instance? Sharing characters can be done by exporting to PNG and sending to whoever you like, but adding a more robust way to do this within the app is something i will get around to doing eventually.I could potentially add in a way for users to access their characters on a cloud service but any sort of accounts would definitely remain optional.
